common/wonder_upgrades/wc_wonder_upgrades.txt

```diff
@@ -275,4 +315,24 @@ upgrade_black_obelisks = {
 	owner_modifier = {
 		same_religion_opinion = 5
 	}
-}
\ No newline at end of file
+}
+upgrade_condemned_halls = {
+	picture = GFX_upgrade_icon_book_red
+
+	potential = { }
+	allow = { }
+	active = { }
+
+	flags = {
+		is_plural
+	}
+	build_time = 40
+	gold_cost_flat = 470
+	gold_cost_ticking = 1
+	local_modifier = {
+		thralls = 175
```

Comment:
Will give any holding in province a 225 special troops.
Make sure that scourge holdings have only thralls and no other special troops.